exchanges/coinbene/coinbene.go

```diff
@@ -0,0 +1,316 @@
+package coinbene
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/config"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/websocket/wshandler"
+	log "github.com/thrasher-corp/gocryptotrader/logger"
+)
+
+// Coinbene is the overarching type across this package
+type Coinbene struct {
+	exchange.Base
+}
+
+const (
+	coinbeneAPIURL     = "http://openapi-exchange.coinbene.com/api/exchange/"
+	coinbeneAuthPath   = "/api/exchange/v2"
+	coinbeneAPIVersion = "v2"
+	buy                = "buy"
+	sell               = "sell"
+
+	// Public endpoints
+	coinbeneFetchTicker    = "/market/ticker/one"
+	coinbeneFetchOrderBook = "/market/orderBook"
+	coinbeneGetTrades      = "/market/trades"
+	coinbeneGetAllPairs    = "/market/tradePair/list"
+	coinbenePairInfo       = "/market/tradePair/one"
+
+	// Authenticated endpoints
+	coinbeneGetUserBalance = "/account/list"
+	coinbenePlaceOrder     = "/order/place"
+	coinbeneOrderInfo      = "/order/info"
+	coinbeneRemoveOrder    = "/order/cancel"
+	coinbeneOpenOrders     = "/order/openOrders"
+	coinbeneClosedOrders   = "/order/closedOrders"
+)
+
+// SetDefaults sets the basic defaults for Coinbene
+func (c *Coinbene) SetDefaults() {
+	c.Name = "Coinbene"
+	c.Enabled = false
+	c.Verbose = false
+	c.RESTPollingDelay = 10
+	c.RequestCurrencyPairFormat.Delimiter = ""
+	c.RequestCurrencyPairFormat.Uppercase = true
+	c.ConfigCurrencyPairFormat.Delimiter = "/"
+	c.ConfigCurrencyPairFormat.Uppercase = true
+	c.AssetTypes = []string{ticker.Spot}
+	c.SupportsAutoPairUpdating = true
+	c.SupportsRESTTickerBatching = false
+	c.Requester = request.New(c.Name,
+		request.NewRateLimit(time.Second, 0),
```

Comment:
Coinbene does have a rate limit. 5 per 2 seconds for auth, and 20 per 2 second for unauth.
This can be written like
`request.NewRateLimit(time.Second, unauthRateLimit ),`
`request.NewRateLimit(time.Minute, authRateLimit),`
Add consts `unauthRateLimit = 10` and `authRateLimit = 300`